build-recipe-mkosi

```diff
@@ -89,6 +89,16 @@ recipe_build_mkosi() {
         fi
     fi
 
+    # If we are running with the obs_scm service, the sources will be unpacked in a directory
+    # in SOURCES/, so try to find the right one, by checking where the top level mkosi.conf is.
+    workdir="$TOPDIR/SOURCES/"
+    for d in "$TOPDIR"/SOURCES/*.obsinfo; do
+        if [ ! -d "${d%.obsinfo}" ]; then continue; fi
+        if [ ! -f "${d%.obsinfo}/mkosi.conf" ]; then continue; fi
+        workdir="${d%.obsinfo}"
```

Comment:
please do not relay on .obsinfo files. You can also have directories via other mechanics. It will esp. break when using native git.

Just look for any directory with a mkosi.conf.